raiden/network/transport/udp/udp_transport.py

```diff
@@ -204,15 +210,27 @@ def start(
             self.init_queue_for(queue_identifier, encoded_queue)
 
         self.server.start()
+        super().start()
+
+    def run(self):
+        """ Runnable main method, perform wait on long-running subtasks """
+        try:
+            self.stop_event.wait()
+        finally:
+            self.stop()  # noop on double call, ensure cleanup and wait on subtasks
+
+    def stop(self, timeout=None):
+        if self.stop_event.ready():
+            return  # double call, happens on normal stop, ignore
```

Comment:
why is `stop` called twice?

Reply:
4 lines above: if an exception happens, I want `run` to call `stop` on exit, to clean up tasks, but it'll also be run on normal `stop`, but second one will be noop